### PR TITLE
Allow `make test` to pass on OSX

### PR DESCRIPTION
**Pull request by spiffxp**
**What this PR does / why we need it**: `make test` doesn't pass on my OSX setup (10.11.6, go1.7, docker 1.13.1) on `master`, `release-1.5`, nor `release-1.4`.  Our [docs on unit tests](https://github.com/kubernetes/community/blob/master/contributors/devel/testing.md#unit-tests) say they should always pass on OS X.  This PR allows them to pass.

**Release note**:
```release-note
NONE
```

ref: #24717 for the motivation behind dereferencing mount symlinks

/cc @kubernetes/sig-testing-pr-reviews

### CODE REVIEW

File: hack/e2e_test.go [hunk 1]
```diff
@@ -314,7 +314,7 @@ func TestGetKubetest(t *testing.T) {
 				if p != c.stat {
 					return nil, fmt.Errorf("Failed to find %s", p)
 				}
-				return FileInfo{time.Now().Add(c.age)}, nil
+				return FileInfo{time.Now().Add(c.age * -1)}, nil
```
Comment (spxtr) [inline hack/e2e_test.go]: Ha! Nice catch.

Comment (rmmh) [inline hack/e2e_test.go]: This should use `Sub(c.age)` instead.

**Comment by spxtr** [comment]
@fejta is on vacation for a while. `hack/` changes look good to me.

/approve

File: hack/e2e_test.go (hunk 1 above)
Comment (spiffxp) [inline hack/e2e_test.go]: Sub takes a time. Age is a duration

**Comment by spiffxp** [comment]
pinging @kubernetes/sig-storage-pr-reviews for the volume-related test fixes

pinging @thockin for approval if this isn't in @lavalamp's wheelhouse

**Pull request merged**